### ISSUE TITLE
Question: Shape Predictor and rotated objects

### ISSUE THREAD

**Issue by kassem-tohme** [davisking/dlib#2080]
Hello,

this is more like a question than an issue or bug.

I have an object where I want to track several keypoints. For the actual object detection I'm using tensorflow, which works quite good and also detects the object when its rotated in scene. I then use these information (rectangle + image) to feed into the dlib shape predictor to get the keypoints of this object. This works quite well when I train a model where the object is not rotated, but performance on images with rotated object is quite bad.

Now my questions are:
1. Is dlib's shape predictor not made for objects which are rotated?
2. Would it help if I use rotated images/labels during training?

Thanks

**Comment by davisking** [comment]
It works well for objects that rotate, but you need to include such examples in your training data.
